templates/common/_base/files/mtu-migration-dispatcher.yaml

```diff
@@ -56,17 +56,19 @@ contents:
         local dev=$1
         local mtu=$2
 
-        dev_routes=$(ip route show dev $dev)
-
-        if [ -n "$dev_routes" ]; then
-            while IFS= read -r line; do
-                current_mtu=$(echo $line | sed -n -e 's/^.*mtu \([0-9]\+\).*/\1/p')
-                echo "mtu=$current_mtu in this route: $line"
-                if [ -z "$current_mtu" ] || [ $current_mtu -gt $mtu ]; then
-                    ip route change ${line} dev $dev mtu ${mtu}
-                fi
-            done <<< "$dev_routes"
-        fi
+        for ip_str in "-4" "-6"; do
+            dev_routes=$(ip $ip_str route show dev $dev)
+
+            if [ -n "$dev_routes" ]; then
+                while IFS= read -r line; do
+                    current_mtu=$(echo $line | sed -n -e 's/^.*mtu \([0-9]\+\).*/\1/p')
+                    echo "mtu=$current_mtu in this route: $line"
+                    if [ -z "$current_mtu" ] || [ $current_mtu -gt $mtu ]; then
+                        ip $ip_str route change ${line} dev $dev mtu ${mtu}
```

Comment:
After some tests, routes continue to be acquired by RA after this runs overriding all the route modifications that we make. So we will need a bit more sophisticated handling here where we basically add our own routes instead of changing them. So:

* Let's set our routes with `proto static`
* Let's set our routes with the highest priority metric. That's `0` for `ipv4` and `1` for ipv6.
* Let's add a route instead of changing it, unless a static route with the same metric exists, in which case we do change it. The metric is part of a route's key.

This piece of code seemed to work well for me in tests:
```
for ip_str in "-4" "-6"; do
    dev_routes=$(ip $ip_str route show dev $dev)
    
    if [ -n "$dev_routes" ]; then
        while IFS= read -r line; do
            route_mtu=$(echo $line | sed -n -e 's/^.*mtu \([0-9]\+\).*/\1/p')
	    route_raw=$(echo $line | sed -e 's/ \(proto\|metric\|mtu\) [[:alnum:]]\+//g')
            [ "$ip_str" = "-4" ] && route_metric=0 || route_metric=1
            echo "mtu=${route_mtu:-undefined} in this route: $line"
            if [ -z "$route_mtu" ] || [ $route_mtu -gt $mtu ]; then
		route="${route_raw} dev $dev proto static metric ${route_metric}"
		echo "Adding or replacing route: ${route}"
		ip $ip_str route replace ${route}
            fi
        done <<< "$dev_routes"
    fi
done

```

Reply:
Thanks! I changed your snippet above in order to better support the case where we have multiple routes for the same destination. 

Also, compared to a commit I force-pushed earlier today, the dispatcher script is run again upon pre-up events. The environment variables we are interested in from Network Manager are available for both pre-up and up events and I saw no difference in the output when executing the dispatcher script either way. We will look into it in a separate PR if needed.